enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,782 @@
+---
+title: converting-machine-api-to-cluster-api
+authors:
+  - "@JoelSpeed"
+reviewers:
+  - "@damdo" # Cluster API and Machine API maintainer
+  - "@nrb" # Cluster API and Machine API maintainer
+  - "@mdbooth" # OpenStack Cluster API and Machine API maintainer
+  - "2uasimojo" # Hive maintainer
+approvers: 
+  - "@vincepri" # Cluster API maintainer within OpenShift
+api-approvers:
+  - "@deads2k"
+creation-date: 2023-08-30
+last-updated: 2024-02-02
+tracking-link: 
+  - https://issues.redhat.com/browse/OCPCLOUD-1578
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Converting Machine API resources to Cluster API
+
+## Summary
+
+To enable OpenShift to migrate from the OpenShift Machine API to the upstream Kubernetes Cluster API project,
+we will need to create a mechanism that allows users to continue using the existing API,
+while we internally migrate to using Cluster API controllers.
+We will also promote, via the addition of newer features, users to migrate to using the Cluster API directly,
+which in turn will allow us to deprecate and eventually, remove (TBC[^1]) the Machine API.
+
+This document outlines the proposed method for that migration process.
+This includes how Machines, MachineSets, MachineHealthChecks, the ControlPlaneMachineSet and the Cluster Autoscaler will be migrated.
+
+[^1]: To remove the Machine API we must prove that only a suitably small number of our customers are still using the API to manage their machines.
+For example, if the number was suitably small that we could reach out to each customer individually to help them move, then it would be feasible.
+For the purposes of this document and this project, we assume that the Machine API will exist in OpenShift 4 forever,
+even though the backing implementation will move to Cluster API.
+In a future iteration, `ClusterFleetEvaluation`` may be used to determine the number of clusters that are still using the Machine API.
+
+## Motivation
+
+The motivation for this project is primarily outlined in this [RFC][rfc-migration-to-capi].
+
+To summarise, we believe that the long term sustainability of OpenShift depends on migrating from Machine API to Cluster API.
+By leveraging the upstream Cluster API project, we will gain access to a community of users and developers, 
+with which we can all work together towards the goal of Kubernetes native Node lifecycle management.
```

Comment:
Another reason we can add:

Having the entire lifecycle of an OpenShift cluster on a single and well-established platform brings in reusability, collaboration, and a more streamlined interface. We can also mention HyperShift uses of Cluster API internally, clients using ROSA through CAPA, and installer re-platforming